diff --git a/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php b/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php
index 1f5026bb6..606fd14eb 100644
--- a/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php
+++ b/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php
@@ -11,6 +11,7 @@
 use Doctrine\ORM\Internal\SQLResultCasing;
 use Doctrine\ORM\Mapping\ClassMetadata;
 use Doctrine\ORM\Utility\PersisterHelper;
+use Ramsey\Uuid\UuidInterface;

 use function array_combine;
 use function implode;
@@ -244,6 +245,15 @@
     public function delete($entity)
     {
         $identifier = $this->em->getUnitOfWork()->getEntityIdentifier($entity);
+
+        foreach ($identifier as $key=>$value) {
+            if ($value instanceof UuidInterface) {
+                $identifier[$key] = $value->getBytes();
+            } else {
+                $identifier[$key] = $value;
+            }
+        }
+
         $id         = array_combine($this->class->getIdentifierColumnNames(), $identifier);
         $types      = $this->getClassIdentifiersTypes($this->class);