diff --git a/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php b/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php
index 098d1a4cd..05791510c 100644
--- a/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php
+++ b/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php
@@ -31,6 +31,7 @@
 use Doctrine\ORM\Utility\IdentifierFlattener;
 use Doctrine\ORM\Utility\PersisterHelper;
 use LengthException;
+use Ramsey\Uuid\UuidInterface;

 use function array_combine;
 use function array_keys;
@@ -590,6 +591,15 @@
     {
         $class      = $this->class;
         $identifier = $this->em->getUnitOfWork()->getEntityIdentifier($entity);
+
+        foreach ($identifier as $key=>$value) {
+            if ($value instanceof UuidInterface) {
+                $identifier[$key] = $value->getBytes();
+            } else {
+                $identifier[$key] = $value;
+            }
+        }
+
         $tableName  = $this->quoteStrategy->getTableName($class, $this->platform);
         $idColumns  = $this->quoteStrategy->getIdentifierColumnNames($class, $this->platform);
         $id         = array_combine($idColumns, $identifier);
